### ISSUE TITLE
Update provisioning and deployment algorithms with additional telemetry

### ISSUE THREAD

**Issue by GeoffCoxMSFT** [microsoft/BotFramework-Composer#8066]
Once the service layer is in place, we will have the atomic blocks to call from the provisioning and deployment algorithms.
We should add telemetry around each action the algorithms take including duration.

**Comment by GeoffCoxMSFT** [comment]
@natalgar - I'll take this to see what I can do in the existing extension for R14.  I think we'll have better telemetry in the new extension as you factor things.

**Comment by GeoffCoxMSFT** [comment]
The extension on the node side only has the logger, not the telemetry client.  We'll need to add the telemetry client to be usable on the node side of extensions before we can add server-side telemetry to provisioning and publishing.  This was a P0 for R13, but we fixed the issue that was causing insufficient error information to be returned, so now this is a lower priority need.
